spirv_msl.cpp

```diff
@@ -4422,6 +4645,59 @@ bool CompilerMSL::skip_argument(uint32_t) const
 	return false;
 }
 
+void CompilerMSL::analyze_image_and_sampler_usage()
```

Comment:
Overriding analyze_image_and_sampler_usage seems a bit off here. This is never called from a parent class (i.e. it would need to be overridden). It seems more direct to move the swizzle_texture_samples path to its own member function in CompilerMSL, and avoid the virtual. CompilerMSL::compile() would then just call analyze_image_and_sampler_usage followed by this one.